compose/material/material/src/desktopMain/kotlin/androidx/compose/material/DesktopMenu.desktop.kt

```diff
@@ -228,29 +296,78 @@ fun CursorDropdownMenu(
     expandedStates.targetState = expanded
 
     if (expandedStates.currentState || expandedStates.targetState) {
-        val transformOriginState = remember { mutableStateOf(TransformOrigin.Center) }
+        OpenDropdownMenu(
+            expandedStates = expandedStates,
+            popupPositionProvider = rememberCursorPositionProvider(),
+            onDismissRequest = onDismissRequest,
+            focusable = focusable,
+            modifier = modifier,
+            content = content
+        )
+    }
+}
 
-        var focusManager: FocusManager? by mutableStateOf(null)
-        var inputModeManager: InputModeManager? by mutableStateOf(null)
+/**
+ * Represents the open/closed state of a dropdown menu.
+ */
+@Stable
+class DropdownMenuState(initialStatus: Status = Status.Closed) {
 
-        Popup(
-            focusable = focusable,
-            onDismissRequest = onDismissRequest,
-            popupPositionProvider = rememberCursorPositionProvider(),
-            onKeyEvent = {
-                handlePopupOnKeyEvent(it, onDismissRequest, focusManager!!, inputModeManager!!)
-            },
-        ) {
-            focusManager = LocalFocusManager.current
-            inputModeManager = LocalInputModeManager.current
-
-            DropdownMenuContent(
-                expandedStates = expandedStates,
-                transformOriginState = transformOriginState,
-                modifier = modifier,
-                content = content
-            )
+    /**
+     * The current status of the menu.
+     */
+    var status: Status by mutableStateOf(initialStatus)
```

Comment:
Is it intentional that it can be changed outside as well, so that is not controlled only by focus and stuff?

Reply:
Is there a reason to limit it?